erpnext/hr/doctype/employee_advance/employee_advance.py

```diff
@@ -53,6 +53,10 @@ def set_total_advance_paid(self):
 		self.set_status()
 		frappe.db.set_value("Employee Advance", self.name , "status", self.status)
 
+	def get_due_advance_amount(self):
```

Comment:
Make it a whitelisted function. If we write it inside the class, then we need to pass the entire doc to the server, which is unnecessary. We only need employee and posting date.